### ISSUE TITLE
Collision on 3D Z axis dose not work

### ISSUE THREAD

**Issue by gabrielzv1233** [4ian/GDevelop#5449]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/4ian/GDevelop/issues)

### Describe the bug

when you try to use collision detection on a 3D object it will detect as in a collision, even if the objects are 30 px apart.
an extra little thing also is separate object action also doesn't allow moving over a wall, or object.

### Steps to reproduce

controls:
W A S D (left right up down for camera)
Left Right Up Down (left right up down for red object, hold Left alt for pixel perfect movement)
Left shift Space (fly camera up fly camera down)
there are collision indicators for the red and blue objects in the top left,
what do do:
move the red object around the blue object
try to move the red object over the walls (grey objects with arrows)
[TryItYourSelf.zip](https://github.com/4ian/GDevelop/files/11898239/TryItYourSelf.zip)

### GDevelop platform

Desktop

### GDevelop version

5.2.166

### Platform info

<details>
*OS Windows*
>

*OS Version: Windows 10 Version 22H2 (OS Build 19045.3086+*
>
</details>

### Additional context

as far as i know, this is a problem on all other versions of GDevelop (IOS, Android, Mac, Ect)

**Comment by AlexandreSi** [comment]
Indeed, nothing has been done yet regarding collisions (with the physics engine or not) in the Z axis. We will come to it at some point.

**Comment by sursonrez** [comment]
@AlexandreSi could that be soon?
